components/builder-api-client/src/lib.rs

```diff
@@ -0,0 +1,160 @@
+#[macro_use]
+extern crate log;
+extern crate regex;
+extern crate serde;
+#[macro_use]
+extern crate serde_derive;
+#[macro_use]
+extern crate serde_json;
+extern crate url;
+
+pub mod error;
+pub use error::{Error, Result};
+
+use std::io::Read;
+use std::path::Path;
+
+use hab_core::package::PackageIdent;
+use hab_http::ApiClient;
+use hyper::client::{IntoUrl, Response, RequestBuilder};
+use hyper::header::{Accept, Authorization, Bearer, ContentType};
+use hyper::status::StatusCode;
+use regex::Regex;
+
+pub struct Client(ApiClient);
+
+#[derive(Deserialize)]
+pub struct ReverseDependencies {
+    pub origin: String,
+    pub name: String,
+    pub rdeps: Vec<String>,
+}
+
+impl Client {
+    pub fn new<U>(
+        depot_url: U,
+        product: &str,
+        version: &str,
+        fs_root_path: Option<&Path>,
+    ) -> Result<Self>
+    where
+        U: IntoUrl,
+    {
+        let re = Regex::new(r"depot/?$").unwrap();
+        let mut url = depot_url.into_url().map_err(Error::URL)?;
+
+        if re.is_match(url.path()) {
+            let mut psm = url.path_segments_mut().unwrap();
+            psm.pop();
+        }
+
+        let api_url = url.as_str();
+
+        Ok(Client(
+            ApiClient::new(api_url, product, version, fs_root_path)
+                .map_err(Error::HabitatHttpClient)?,
+        ))
+    }
+
+    /// Create a job.
+    ///
+    /// # Failures
+    ///
+    /// * Remote API Server is not available
+    ///
+    /// # Panics
+    ///
+    /// * Authorization token was not set on client
+    pub fn create_job(&self, ident: &PackageIdent, token: &str) -> Result<(String)> {
+        debug!("Creating a job for {}", ident);
+
+        let body = json!({
+            "project_id": format!("{}", ident)
+        });
+
+        let sbody = serde_json::to_string(&body).unwrap();
+
+        let result = self.add_authz(self.0.post("jobs"), token)
+            .body(&sbody)
+            .header(Accept::json())
+            .header(ContentType::json())
+            .send();
+        match result {
+            Ok(mut response) => {
+                match response.status {
+                    StatusCode::Created => {
+                        let mut encoded = String::new();
+                        response.read_to_string(&mut encoded).map_err(Error::IO)?;
+                        debug!("Body: {:?}", encoded);
+                        let v: serde_json::Value =
+                            serde_json::from_str(&encoded).map_err(Error::JSON)?;
+                        let id = v["id"].as_str().unwrap();
+                        Ok(id.to_string())
+                    }
+                    StatusCode::Unauthorized => {
+                        Err(Error::APIError(
+                            response.status,
+                            "Your GitHub token requires both user:email and read:org \
+                                             permissions."
+                                .to_string(),
```

Comment:
There are some (predictable) commonalities between the `builder-api-client` and `builder-depot-client` crates, given the similarities of what they're doing.  I can extract out some of the common stuff into a shared crate, but it might be worth considering how much time we want to spend refactoring this stuff when it will likely be getting a major overhaul anyway when 1) everything is consolidated from the depot architecture refactor and 2) the upgrade to hyper 0.11.  I don't think we want to spend _too_ much time polishing this stuff at this point.